services/src/main/java/org/fao/geonet/api/pages/PagesAPI.java

```diff
@@ -399,14 +429,37 @@ public ResponseEntity<List<org.fao.geonet.api.pages.PageProperties>> listPages(
                 || page.getStatus().equals(Page.PageStatus.PUBLIC)
                 || page.getStatus().equals(Page.PageStatus.PUBLIC_ONLY) && !us.isAuthenticated()) {
                 if (section == null) {
-                    filteredResult.add(new org.fao.geonet.api.pages.PageProperties(page));
+                    filteredResult.add(new PageProperties(page));
                 } else {
                     final List<Page.PageSection> sections = page.getSections();
                     final boolean containsRequestedSection = sections.contains(section);
                     if (containsRequestedSection) {
-                        filteredResult.add(new org.fao.geonet.api.pages.PageProperties(page));
+                        filteredResult.add(new PageProperties(page));
                     }
                 }
+            } else if (page.getStatus().equals(Page.PageStatus.GROUPS) && us.getProfile() != null && us.getProfile() != Profile.Guest) {
```

Comment:
This logic seems to be missing the section filtering logic that exists in the previous block. It seems like if the api is called based on group that it will always return all the sections.

I believe your logic should go in the 
`if (containsRequestedSection) {`
block above.

Reply:
May also be good to call a function specifically for the group handling - it may make it cleaner.

Reply:
Refactored the logic to a private method. I also add such condition to the block of condition with other status. It looks more organized